docker/nexus/Dockerfile

```
@@ -0,0 +1,43 @@
+FROM python:3.10-slim AS builder
+
+ARG INSTALL_RUST=false
+
+
+WORKDIR /app
+
+COPY . .
+
+RUN if [ "$INSTALL_RUST" = "true" ]; then \
+        apt-get update && apt-get install -y --no-install-recommends curl build-essential && \ 
+        curl https://sh.rustup.rs -sSf | sh -s -- -y && \
+        . $HOME/.cargo/env && \ 
+        rustup update && \
+        rustup default stable; \
+    fi
```

We can make it look like a shell script.

```suggestion
RUN <<EOF
  set -eo pipefail
  
  if [ "$INSTALL_RUST" = "true" ]; then
     apt-get update
     apt-get install -y --no-install-recommends curl build-essential
     curl https://sh.rustup.rs -sSf | sh -s -- -y
     . $HOME/.cargo/env
     rustup update
     rustup default stable
  fi
EOF
```